### PR TITLE
executor: refine impl of IndexMergeReader

### PR DESCRIPTION
**Pull request by sduzh**
### What problem does this PR solve? 

Refine the implementation of IndexMergeReader

### Check List 

Tests 

 - Unit test
 - Integration test

Code changes

Side effects

 - Possible performance regression
 - Increased code complexity

Related changes

Release note

### CODE REVIEW

File: executor/index_merge_reader.go [hunk 1]
```diff
@@ -711,3 +589,72 @@ func (w *indexMergeTableScanWorker) executeTask(ctx context.Context, task *looku
 	}
 	return nil
 }
+
+type handleMerger interface {
```
Comment (shenli) [inline executor/index_merge_reader.go]: Could you add some comments for the following two structs?

Comment (sduzh) [reply executor/index_merge_reader.go]: updated

**Comment by sduzh** [comment]
@XuHuaiyu PTAL

**Comment by XuHuaiyu** [comment]
Discussed offline, new PR will be pushed.

**Pull request closed without merging**